scipy/stats/_qmc.py

```diff
@@ -306,6 +308,39 @@ def discrepancy(
                          f" {set(methods)!r}")
 
 
+def geometric_discrepancy(
+        sample: npt.ArrayLike,
+        method: Literal["mindist", "mst"] = "mindist",
+        metric: str = "euclidean") -> float:
+    """Discrepancy of a given sample based on geometric metrics.
+
+    Parameters
+    ----------
+    sample : array_like (n, d)
+        The sample to compute the discrepancy from.
+    method: str, optional
+        The method to use. One of ``mindist`` for minimum distance or ``mst`` for minimum spanning tree.
+    metric: str or function, optional
+        The distance metric to use. See the documentation for ``scipy.spatial.distance.pdist``.
+
+    Returns
+    -------
+    discrepancy : float
+        Discrepancy.
+    """
```

Comment:
This could use a small Notes section explaining a bit the metrics, an Example section with 1-2 examples (bonus point if you make examples with the tree for mst and packing circles for mindist. I have a gist you can use: https://gist.github.com/tupui/2719245158f2dc2366a9256b58a85519). And lastly can you add a reference section with 1-2 articles?

Reply:
Added notes, references and examples.

> bonus point if you make examples with the tree for mst and packing circles for mindist.

Do you mean examples with specific arrangements of points?

Reply:
Thank you!

For the figures, I was thinking about things like that:

This one is the output of the gist I shared
![43350899-9efe0f08-920b-11e8-8a20-e9dab98f7a52](https://github.com/scipy/scipy/assets/23188539/25667333-fb1e-452f-ab6b-6bef3dd0684a)

And this is a maximin on the left and a minimax sampling on the right. Here I am just saying we could have a circle around the "critical point" to show the minimal distance (the circle would touch the nearest point then).
![Figure-Maximin-left-and-Minimax-right-designs_W640](https://github.com/scipy/scipy/assets/23188539/07f8dfe6-df19-486e-93e6-edeeb2582dac)

Reply:
Nice 👍

I was actually thinking about 2 different figures, one per method. But with a legend, having both on the same figure could work and save some space.

I would set the aspect ratio to 1 so that the circle is really a circle.